accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,395 @@
+- Feature Name: Container as a Service Platform integration with Uyuni/SUSE Manager
+- Start Date: 2019-08-05
+- RFC PR: https://github.com/uyuni-project/uyuni-rfc/pull/10
+
+# Summary
+[summary]: #summary
+
+This RFC describes how Uyuni/SUSE Manager can be leveraged to perform cross-product integration with a cluster solution like Container as a Service Platform.
+
+# Motivation
+[motivation]: #motivation
+
+<!-- - Why are we doing this?
+- What use cases does it support?
+- What is the expected outcome?
+
+Describe the problem you are trying to solve, and its constraints, without coupling them too closely to the solution you have in mind. If this RFC is not accepted, the motivation can be used to develop alternative solutions. -->
+
+Uyuni/SUSE Manager is the systems management tool and must offer to the end-user the possibility to deploy and manage other SUSE products like CaaS Platform.
+
+We will be focusing first on the following general scenarios:
+
+1. Deploy a cluster using Uyuni/SUSE Manager
+2. Manage a cluster using Uyuni/SUSE Manager
+
+The two subsequent CaaS Platform-specific sections will follow.
+
+# Detailed design
+[design]: #detailed-design
+
+<!-- This is the bulk of the RFC. Explain the design in enough detail for somebody familiar with the product to understand, and for somebody familiar with the internals to implement.
+
+This section should cover architecture aspects and the rationale behind disruptive technical decisions (when applicable), as well as corner-cases and warnings. Whenever the new feature creates new user interactions, this section should include examples of how the feature will be used. -->
+
+## Generic design: integrate Uyuni/SUSE Manager with a cluster provider
+
+### Deploy a cluster using Uyuni/SUSE Manager
+
+Under Uyuni/SUSE Manager main menu, a new menu item "Clusters" should be introduced.
+All deployable and manageable cluster products (e.g. CaaSP, SOC, SES) will be in the sub-menu.
+
+Under that item, it should be possible to create, edit, and remove a cluster.
+
+To create a cluster, a user has to enter its name.
+
+Every detail about the cluster is saved into corresponding database tables (e.g. `susecluster`).
```

Comment:
I'm not sure whether this approach will scale. Yes, there are a few things that will be identical for all types of clusters, like the name SUSE Manager knows it as. And those should probably be in the DB.
But cluster managers can be implemented in many ways and may require a variety of different configuration options. So I was rather thinking of implementing the GUI part in a pluggable way as well, similar to what we're doing with Salt Formulas with Forms. And for storing that additional cluster-specific data, I think we'll have to look into "schemaless" or dynamic schemas (e.g. a set of YAML or JSON files). Otherwise you'll have to do DB schema updates every time something changes in one of the cluster stacks. Whether those files are then stored in PostgreSQL or kept external, e.g. in the filesystem or Git, is another question.

Reply:
Yes, I agree. I added a section to specify this additional information saving.